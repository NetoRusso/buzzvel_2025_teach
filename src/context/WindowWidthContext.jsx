'use client'
import { createContext, useState, useEffect } from "react";

const WindowWidthContext = createContext();

export const WindowWidthProvider = ({ children }) => {
  const [windowWidth, setWindowWidth] = useState(0);
  const [isMobile, setIsMobile] = useState(false);

  useEffect(() => {
    const handleResize = () => {
      setWindowWidth(window.innerWidth);
    }

    window.addEventListener("resize", handleResize);

    handleResize();

    return () => {
      window.removeEventListener("resize", handleResize);
    };
  }, []);

  useEffect(() => {
    if(windowWidth < 840) {
      setIsMobile(true);

    } else if(windowWidth > 840) {
      setIsMobile(false);

    }
  }, [windowWidth]);

  return (
    <WindowWidthContext.Provider value={{ windowWidth, isMobile }}>
      {children}
    </WindowWidthContext.Provider>
  );
};

export { WindowWidthContext };