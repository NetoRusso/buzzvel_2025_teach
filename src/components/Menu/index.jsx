'use client'
import styles from "./styles.module.css";
import { useState, useRef, useEffect, useContext } from "react";
import Image from "next/image";
import Link from "next/link";
import classNames from "classnames";

import Logo from "@/assets/Logo-colored.svg";
import arrow from "@/assets/arrow_down.svg";
import arrowLeft from "@/assets/arrow_left.svg";
import { WindowWidthContext } from "@/context/WindowWidthContext";


const Menu = () => {
  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
  const [modal, setModal] = useState(false);
  const [menuOpen, setMenuOpen] = useState(false);
  const dropdownRef = useRef(null);

  const { windowWidth, isMobile } = useContext(WindowWidthContext);

  useEffect(() => {
    setIsDropdownVisible(false);
    setModal(false);
    setMenuOpen(false);
  }, [windowWidth]);

  const handleClick = () => {
    setIsDropdownVisible(!isDropdownVisible);
  };

  const handleClose = () => {
    setModal(false);
    setIsDropdownVisible(false);
    setTimeout(() => {
      setMenuOpen(false);
    }, 500);
  }

  return (
    <div className={styles.container}>
      <div className={classNames({
        [styles.content]: true,
        [styles.content_mobile]: isMobile,
      })}>
        <Image
          src={Logo}
          alt="Logo"
          width={103}
          height={35}
          className={styles.menu_logo}
          onClick={() => {
            window.scrollTo({ top: 0, behavior: 'smooth' });
          }}
        />
        {!isMobile ? (
          <div className={styles.menu_desktop}>
            <nav className={styles.menu_desktop_nav}>
              <ul className={styles.menu_desktop_nav_list}>
                <li className={styles.menu_desktop_nav_list_item}>
                  <Link href="#" className={styles.menu_desktop_nav_list_item_link}>Products</Link>
                </li>
                <li className={styles.menu_desktop_nav_list_item}>
                  <Link href="#" className={styles.menu_desktop_nav_list_item_link}>Solutions</Link>
                </li>
                <li className={styles.menu_desktop_nav_list_item}>
                  <Link href="#" className={styles.menu_desktop_nav_list_item_link}>Pricing</Link>
                </li>
                <li
                  className={classNames({
                    [styles.menu_desktop_nav_list_item]: true,
                    [styles.menu_desktop_nav_list_item_dropdown]: true,
                  })}
                  onClick={handleClick}
                >
                  <p
                    className={classNames({
                      [styles.menu_desktop_nav_list_item_link]: true,
                      [styles.menu_desktop_nav_list_item_link_open_dropdown]: true,
                    })}
                  >
                    Resources
                    <Image src={arrow} alt="arrow" width={13.34} height={7.67} />
                  </p>
                  <div
                    ref={dropdownRef}
                    className={classNames({
                      [styles.menu_desktop_nav_list_item_link_dropdown]: true,
                      [styles.active]: isDropdownVisible,
                    })}
                  >
                    <ul className={styles.menu_desktop_nav_list_item_link_dropdown_list}>
                      <li className={styles.menu_desktop_nav_dropdown_list_item}>
                        <Link href="#" className={styles.menu_desktop_nav_dropdown_list_item_link}>
                          Help Center
                        </Link>
                      </li>
                      <li className={styles.menu_desktop_nav_dropdown_list_item}>
                        <Link href="#" className={styles.menu_desktop_nav_dropdown_list_item_link}>
                          Blog
                        </Link>
                      </li>
                      <li className={styles.menu_desktop_nav_dropdown_list_item}>
                        <Link href="#" className={styles.menu_desktop_nav_dropdown_list_item_link}>
                          Tutorials
                        </Link>
                      </li>
                      <li className={styles.menu_desktop_nav_dropdown_list_item}>
                        <Link href="#" className={styles.menu_desktop_nav_dropdown_list_item_link}>
                          FAQs
                        </Link>
                      </li>
                    </ul>
                  </div>
                </li>
              </ul>
            </nav>

            <div className={styles.menu_desktop_acess}>
              <Link href="#" className={styles.menu_desktop_nav_list_item_link}>
                Log In
              </Link>
              <button className={styles.menu_desktop_acess_button}>
                Sing Up Now
              </button>
            </div>
          </div>
        ) : (
          <>
            {modal && (<div
              className={styles.modal}
              onClick={() => handleClose()}
            />)}
            <div
              className={styles.menu_mobile_btn}
              onClick={() => {
                setModal(true);
                setMenuOpen(true);
              }}
            >
              <span />
              <span />
              <span />
            </div>
            {menuOpen && (
              <div className={classNames({
                [styles.menu_mobile_box]: true,
                [styles.menu_mobile_box_closing]: !modal,
              })}>
                <span className={styles.menu_mobile_close_btn} onClick={() => handleClose()}> &times; </span>
                <div className={styles.menu_mobile_box_content}>
                  <div className={styles.menu_mobile_nav}>
                    <ul className={styles.menu_mobile_nav_list}>
                      <li className={styles.menu_mobile_nav_list_item}>
                        <Link href="#" className={styles.menu_mobile_nav_list_item_link}>Products</Link>
                      </li>
                      <li className={styles.menu_mobile_nav_list_item}>
                        <Link href="#" className={styles.menu_mobile_nav_list_item_link}>Solutions</Link>
                      </li>
                      <li className={styles.menu_mobile_nav_list_item}>
                        <Link href="#" className={styles.menu_mobile_nav_list_item_link}>Pricing</Link>
                      </li>
                      <li className={styles.menu_mobile_nav_list_item}>
                        <p
                          className={styles.menu_mobile_nav_list_item_link}
                          onClick={() => setIsDropdownVisible(!isDropdownVisible)}
                        >
                          <Image src={arrowLeft} alt="seta para esquerda" width={7.67} height={13.34} />
                          Resources
                        </p>
                        {isDropdownVisible && (
                          <div className={styles.menu_mobile_dropdown}>
                            <span className={styles.menu_mobile_dropdown_close_btn} onClick={() => setIsDropdownVisible(!isDropdownVisible)}> &times; </span>
                            <div className={styles.menu_mobile_dropdow_content}>
                              <ul className={styles.menu_mobile_dropdown_list}>
                                <li className={styles.menu_mobile_dropdown_list_item}>
                                  <Link href="#" className={styles.menu_mobile_dropdown_list_item_link}>Help Center</Link>
                                </li>
                                <li className={styles.menu_mobile_dropdown_list_item}>
                                  <Link href="#" className={styles.menu_mobile_dropdown_list_item_link}>Blog</Link>
                                </li>
                                <li className={styles.menu_mobile_dropdown_list_item}>
                                  <Link href="#" className={styles.menu_mobile_dropdown_list_item_link}>Tutorials</Link>
                                </li>
                                <li className={styles.menu_mobile_dropdown_list_item}>
                                  <Link href="#" className={styles.menu_mobile_dropdown_list_item_link}>FAQs</Link>
                                </li>

                              </ul>
                            </div>
                          </div>
                        )}
                      </li>
                      <li className={classNames({
                        [styles.menu_mobile_nav_list_item]: true,
                        [styles.menu_mobile_nav_list_item_login]: true,
                      })}>
                        <Link href="#" className={styles.menu_mobile_nav_list_item_link} >
                          Log In
                        </Link>
                        <button className={styles.menu_desktop_acess_button}>
                          Sing Up Now
                        </button>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            )}
          </>
        )}
      </div>
    </div>
  );
};

export default Menu;