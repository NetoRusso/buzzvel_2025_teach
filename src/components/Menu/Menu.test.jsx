import React from 'react';
import { render, screen, fireEvent, act, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import Menu from './index';
import { WindowWidthContext } from '@/context/WindowWidthContext';

jest.mock('next/image', () => {
  const ImageComponent = (props) => {
    let { src: srcProp, alt, ...rest } = props;
    let finalSrc = srcProp;
    if (typeof srcProp === 'object' && srcProp !== null && srcProp.src) {
      finalSrc = srcProp.src;
    }
    return <img src={finalSrc} alt={alt || ''} {...rest} />;
  };
  ImageComponent.displayName = 'NextImageMock';
  return {
    __esModule: true,
    default: ImageComponent,
  };
});

jest.mock('next/link', () => {
  const LinkComponent = ({ children, href, ...props }) => (
    <a href={href} {...props}>{children}</a>
  );
  LinkComponent.displayName = 'NextLinkMock';
  return {
    __esModule: true,
    default: LinkComponent,
  };
});

global.scrollTo = jest.fn();

const mockWindowWidthContext = (isMobileValue, windowWidthValue = isMobileValue ? 400 : 1024) => ({
  isMobile: isMobileValue,
  windowWidth: windowWidthValue,
  isResizing: false,
});

const renderWithContext = (ui, providerProps) => {
  return render(
    <WindowWidthContext.Provider value={providerProps}>
      {ui}
    </WindowWidthContext.Provider>
  );
};

describe('Menu Component', () => {
  beforeEach(() => {
    jest.useFakeTimers();
    global.scrollTo.mockClear();
  });

  afterEach(() => {
    jest.clearAllTimers();
    jest.useRealTimers();
    jest.restoreAllMocks();
  });

  describe('Desktop View', () => {
    const desktopContextValue = mockWindowWidthContext(false);

    test('renders desktop menu items and logo correctly', () => {
      renderWithContext(<Menu />, desktopContextValue);
      expect(screen.getByAltText('Logo')).toBeInTheDocument();
      expect(screen.getByText('Products')).toBeInTheDocument();
      expect(screen.getByText('Solutions')).toBeInTheDocument();
      expect(screen.getByText('Pricing')).toBeInTheDocument();
      expect(screen.getByTestId('desktop-resources-trigger')).toBeInTheDocument();
      expect(screen.getByText('Log In')).toBeInTheDocument();
      expect(screen.getByRole('button', { name: /Sing Up Now/i })).toBeInTheDocument();
      expect(screen.queryByTestId('mobile-menu-trigger')).not.toBeInTheDocument();
    });

    test('toggles Resources dropdown on click', async () => {
      renderWithContext(<Menu />, desktopContextValue);
      const resourcesTrigger = screen.getByTestId('desktop-resources-trigger');
      const resourcesDropdown = screen.getByTestId('desktop-resources-dropdown');

      expect(resourcesDropdown).not.toHaveClass('active');

      fireEvent.click(resourcesTrigger);
      expect(resourcesDropdown).toHaveClass('active');
      await waitFor(() => {
        expect(screen.getByText('Help Center')).toBeVisible();
      });
      expect(screen.getByText('Blog')).toBeVisible();

      fireEvent.click(resourcesTrigger);
      await waitFor(() => {
        expect(resourcesDropdown).not.toHaveClass('active');
      });
    });

    test('logo click scrolls to top', () => {
      renderWithContext(<Menu />, desktopContextValue);
      fireEvent.click(screen.getByAltText('Logo'));
      expect(global.scrollTo).toHaveBeenCalledWith({ top: 0, behavior: 'smooth' });
    });

    test('matches snapshot on desktop', () => {
      const { container } = renderWithContext(<Menu />, desktopContextValue);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('matches snapshot on desktop with dropdown open', async () => {
      const { container } = renderWithContext(<Menu />, desktopContextValue);
      const resourcesTrigger = screen.getByTestId('desktop-resources-trigger');
      fireEvent.click(resourcesTrigger);
      await waitFor(() => {
        expect(screen.getByTestId('desktop-resources-dropdown')).toHaveClass('active');
      });
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Mobile View', () => {
    const mobileContextValue = mockWindowWidthContext(true);

    test('renders mobile menu trigger and logo correctly', () => {
      renderWithContext(<Menu />, mobileContextValue);
      expect(screen.getByAltText('Logo')).toBeInTheDocument();
      expect(screen.getByTestId('mobile-menu-trigger')).toBeInTheDocument();
      expect(screen.queryByText('Products')).not.toBeInTheDocument();
    });

    test('opens and closes mobile menu', async () => {
      renderWithContext(<Menu />, mobileContextValue);
      const openMenuButton = screen.getByTestId('mobile-menu-trigger');

      expect(screen.queryByTestId('mobile-menu-box')).not.toBeInTheDocument();

      fireEvent.click(openMenuButton);
      const menuBox = await screen.findByTestId('mobile-menu-box');
      expect(menuBox).toBeInTheDocument();
      expect(screen.getByText('Products')).toBeInTheDocument();

      const closeButton = screen.getByTestId('mobile-menu-close-button');
      fireEvent.click(closeButton);

      act(() => {
        jest.advanceTimersByTime(500);
      });

      await waitFor(() => {
        expect(screen.queryByTestId('mobile-menu-box')).not.toBeInTheDocument();
      });
    });

    test('toggles Resources dropdown in mobile menu', async () => {
      renderWithContext(<Menu />, mobileContextValue);
      const openMenuButton = screen.getByTestId('mobile-menu-trigger');
      fireEvent.click(openMenuButton);

      await screen.findByTestId('mobile-menu-box');

      const resourcesTriggerMobile = screen.getByTestId('mobile-resources-trigger');
      expect(screen.queryByTestId('mobile-resources-dropdown')).not.toBeInTheDocument();

      fireEvent.click(resourcesTriggerMobile);
      const resourcesDropdownMobile = await screen.findByTestId('mobile-resources-dropdown');
      expect(resourcesDropdownMobile).toBeInTheDocument();
      expect(screen.getByText('Help Center')).toBeInTheDocument();

      const closeDropdownButton = screen.getByTestId('mobile-resources-dropdown-close-button');
      fireEvent.click(closeDropdownButton);
      expect(screen.queryByTestId('mobile-resources-dropdown')).not.toBeInTheDocument();
    });

    test('closes mobile menu and dropdowns when windowWidth changes (mobile to desktop)', async () => {
      const initialMobileContext = mockWindowWidthContext(true, 400);
      const { rerender } = renderWithContext(<Menu />, initialMobileContext);

      const openMenuButton = screen.getByTestId('mobile-menu-trigger');
      fireEvent.click(openMenuButton);
      await screen.findByTestId('mobile-menu-box');
      const resourcesTriggerMobile = screen.getByTestId('mobile-resources-trigger');
      fireEvent.click(resourcesTriggerMobile);
      await screen.findByTestId('mobile-resources-dropdown');

      expect(screen.getByTestId('mobile-menu-box')).toBeInTheDocument();
      expect(screen.getByTestId('mobile-resources-dropdown')).toBeInTheDocument();

      const newDesktopContext = mockWindowWidthContext(false, 1200);
      rerender(
        <WindowWidthContext.Provider value={newDesktopContext}>
          <Menu />
        </WindowWidthContext.Provider>
      );
      
      await waitFor(() => {
        expect(screen.queryByTestId('mobile-menu-box')).not.toBeInTheDocument();
        expect(screen.queryByTestId('mobile-resources-dropdown')).not.toBeInTheDocument();
        expect(screen.getByText('Products')).toBeInTheDocument();
        expect(screen.getByTestId('desktop-resources-trigger')).toBeInTheDocument();
      });
    });

    test('matches snapshot on mobile (menu closed)', () => {
      const { container } = renderWithContext(<Menu />, mobileContextValue);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('matches snapshot on mobile (menu open)', async () => {
      const { container } = renderWithContext(<Menu />, mobileContextValue);
      const openMenuButton = screen.getByTestId('mobile-menu-trigger');
      fireEvent.click(openMenuButton);
      await screen.findByTestId('mobile-menu-box');
      expect(container.firstChild).toMatchSnapshot();
    });

    test('matches snapshot on mobile (menu and resources dropdown open)', async () => {
        const { container } = renderWithContext(<Menu />, mobileContextValue);
        const openMenuButton = screen.getByTestId('mobile-menu-trigger');
        fireEvent.click(openMenuButton);
        await screen.findByTestId('mobile-menu-box');

        const resourcesTriggerMobile = screen.getByTestId('mobile-resources-trigger');
        fireEvent.click(resourcesTriggerMobile);
        await screen.findByTestId('mobile-resources-dropdown');

        expect(container.firstChild).toMatchSnapshot();
      });
  });
});