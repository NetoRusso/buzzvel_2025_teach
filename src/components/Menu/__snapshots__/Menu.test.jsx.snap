// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component Desktop View matches snapshot on desktop 1`] = `
<div
  class="container"
>
  <div
    class="content"
  >
    <img
      alt="Logo"
      class="menu_logo"
      height="35"
      src="/img.jpg"
      width="103"
    />
    <div
      class="menu_desktop"
    >
      <nav
        class="menu_desktop_nav"
      >
        <ul
          class="menu_desktop_nav_list"
        >
          <li
            class="menu_desktop_nav_list_item"
          >
            <a
              class="menu_desktop_nav_list_item_link"
              href="#"
            >
              Products
            </a>
          </li>
          <li
            class="menu_desktop_nav_list_item"
          >
            <a
              class="menu_desktop_nav_list_item_link"
              href="#"
            >
              Solutions
            </a>
          </li>
          <li
            class="menu_desktop_nav_list_item"
          >
            <a
              class="menu_desktop_nav_list_item_link"
              href="#"
            >
              Pricing
            </a>
          </li>
          <li
            class="menu_desktop_nav_list_item menu_desktop_nav_list_item_dropdown"
            data-testid="desktop-resources-trigger"
          >
            <p
              class="menu_desktop_nav_list_item_link menu_desktop_nav_list_item_link_open_dropdown"
            >
              Resources
              <img
                alt="arrow"
                height="7.67"
                src="/img.jpg"
                width="13.34"
              />
            </p>
            <div
              class="menu_desktop_nav_list_item_link_dropdown"
              data-testid="desktop-resources-dropdown"
            >
              <ul
                class="menu_desktop_nav_list_item_link_dropdown_list"
              >
                <li
                  class="menu_desktop_nav_dropdown_list_item"
                >
                  <a
                    class="menu_desktop_nav_dropdown_list_item_link"
                    href="#"
                  >
                    Help Center
                  </a>
                </li>
                <li
                  class="menu_desktop_nav_dropdown_list_item"
                >
                  <a
                    class="menu_desktop_nav_dropdown_list_item_link"
                    href="#"
                  >
                    Blog
                  </a>
                </li>
                <li
                  class="menu_desktop_nav_dropdown_list_item"
                >
                  <a
                    class="menu_desktop_nav_dropdown_list_item_link"
                    href="#"
                  >
                    Tutorials
                  </a>
                </li>
                <li
                  class="menu_desktop_nav_dropdown_list_item"
                >
                  <a
                    class="menu_desktop_nav_dropdown_list_item_link"
                    href="#"
                  >
                    FAQs
                  </a>
                </li>
              </ul>
            </div>
          </li>
        </ul>
      </nav>
      <div
        class="menu_desktop_acess"
      >
        <a
          class="menu_desktop_nav_list_item_link"
          href="#"
        >
          Log In
        </a>
        <button
          class="menu_desktop_acess_button"
        >
          Sing Up Now
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Menu Component Desktop View matches snapshot on desktop with dropdown open 1`] = `
<div
  class="container"
>
  <div
    class="content"
  >
    <img
      alt="Logo"
      class="menu_logo"
      height="35"
      src="/img.jpg"
      width="103"
    />
    <div
      class="menu_desktop"
    >
      <nav
        class="menu_desktop_nav"
      >
        <ul
          class="menu_desktop_nav_list"
        >
          <li
            class="menu_desktop_nav_list_item"
          >
            <a
              class="menu_desktop_nav_list_item_link"
              href="#"
            >
              Products
            </a>
          </li>
          <li
            class="menu_desktop_nav_list_item"
          >
            <a
              class="menu_desktop_nav_list_item_link"
              href="#"
            >
              Solutions
            </a>
          </li>
          <li
            class="menu_desktop_nav_list_item"
          >
            <a
              class="menu_desktop_nav_list_item_link"
              href="#"
            >
              Pricing
            </a>
          </li>
          <li
            class="menu_desktop_nav_list_item menu_desktop_nav_list_item_dropdown"
            data-testid="desktop-resources-trigger"
          >
            <p
              class="menu_desktop_nav_list_item_link menu_desktop_nav_list_item_link_open_dropdown"
            >
              Resources
              <img
                alt="arrow"
                height="7.67"
                src="/img.jpg"
                width="13.34"
              />
            </p>
            <div
              class="menu_desktop_nav_list_item_link_dropdown active"
              data-testid="desktop-resources-dropdown"
            >
              <ul
                class="menu_desktop_nav_list_item_link_dropdown_list"
              >
                <li
                  class="menu_desktop_nav_dropdown_list_item"
                >
                  <a
                    class="menu_desktop_nav_dropdown_list_item_link"
                    href="#"
                  >
                    Help Center
                  </a>
                </li>
                <li
                  class="menu_desktop_nav_dropdown_list_item"
                >
                  <a
                    class="menu_desktop_nav_dropdown_list_item_link"
                    href="#"
                  >
                    Blog
                  </a>
                </li>
                <li
                  class="menu_desktop_nav_dropdown_list_item"
                >
                  <a
                    class="menu_desktop_nav_dropdown_list_item_link"
                    href="#"
                  >
                    Tutorials
                  </a>
                </li>
                <li
                  class="menu_desktop_nav_dropdown_list_item"
                >
                  <a
                    class="menu_desktop_nav_dropdown_list_item_link"
                    href="#"
                  >
                    FAQs
                  </a>
                </li>
              </ul>
            </div>
          </li>
        </ul>
      </nav>
      <div
        class="menu_desktop_acess"
      >
        <a
          class="menu_desktop_nav_list_item_link"
          href="#"
        >
          Log In
        </a>
        <button
          class="menu_desktop_acess_button"
        >
          Sing Up Now
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Menu Component Mobile View matches snapshot on mobile (menu and resources dropdown open) 1`] = `
<div
  class="container"
>
  <div
    class="content content_mobile"
  >
    <img
      alt="Logo"
      class="menu_logo"
      height="35"
      src="/img.jpg"
      width="103"
    />
    <div
      class="modal"
      data-testid="mobile-menu-modal-backdrop"
    />
    <div
      aria-label="Open mobile menu"
      class="menu_mobile_btn"
      data-testid="mobile-menu-trigger"
      role="button"
    >
      <span />
      <span />
      <span />
    </div>
    <div
      class="menu_mobile_box"
      data-testid="mobile-menu-box"
    >
      <span
        aria-label="Close mobile menu"
        class="menu_mobile_close_btn"
        data-testid="mobile-menu-close-button"
        role="button"
      >
        ×
      </span>
      <div
        class="menu_mobile_box_content"
      >
        <div
          class="menu_mobile_nav"
        >
          <ul
            class="menu_mobile_nav_list"
          >
            <li
              class="menu_mobile_nav_list_item"
            >
              <a
                class="menu_mobile_nav_list_item_link"
                href="#"
              >
                Products
              </a>
            </li>
            <li
              class="menu_mobile_nav_list_item"
            >
              <a
                class="menu_mobile_nav_list_item_link"
                href="#"
              >
                Solutions
              </a>
            </li>
            <li
              class="menu_mobile_nav_list_item"
            >
              <a
                class="menu_mobile_nav_list_item_link"
                href="#"
              >
                Pricing
              </a>
            </li>
            <li
              class="menu_mobile_nav_list_item"
            >
              <p
                class="menu_mobile_nav_list_item_link"
                data-testid="mobile-resources-trigger"
                role="button"
              >
                <img
                  alt="seta para esquerda"
                  height="13.34"
                  src="/img.jpg"
                  width="7.67"
                />
                Resources
              </p>
              <div
                class="menu_mobile_dropdown"
                data-testid="mobile-resources-dropdown"
              >
                <span
                  aria-label="Close resources dropdown"
                  class="menu_mobile_dropdown_close_btn"
                  data-testid="mobile-resources-dropdown-close-button"
                  role="button"
                >
                  ×
                </span>
                <div
                  class="menu_mobile_dropdow_content"
                >
                  <ul
                    class="menu_mobile_dropdown_list"
                  >
                    <li
                      class="menu_mobile_dropdown_list_item"
                    >
                      <a
                        class="menu_mobile_dropdown_list_item_link"
                        href="#"
                      >
                        Help Center
                      </a>
                    </li>
                    <li
                      class="menu_mobile_dropdown_list_item"
                    >
                      <a
                        class="menu_mobile_dropdown_list_item_link"
                        href="#"
                      >
                        Blog
                      </a>
                    </li>
                    <li
                      class="menu_mobile_dropdown_list_item"
                    >
                      <a
                        class="menu_mobile_dropdown_list_item_link"
                        href="#"
                      >
                        Tutorials
                      </a>
                    </li>
                    <li
                      class="menu_mobile_dropdown_list_item"
                    >
                      <a
                        class="menu_mobile_dropdown_list_item_link"
                        href="#"
                      >
                        FAQs
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </li>
            <li
              class="menu_mobile_nav_list_item menu_mobile_nav_list_item_login"
            >
              <a
                class="menu_mobile_nav_list_item_link"
                href="#"
              >
                Log In
              </a>
              <button
                class="menu_desktop_acess_button"
              >
                Sing Up Now
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Menu Component Mobile View matches snapshot on mobile (menu closed) 1`] = `
<div
  class="container"
>
  <div
    class="content content_mobile"
  >
    <img
      alt="Logo"
      class="menu_logo"
      height="35"
      src="/img.jpg"
      width="103"
    />
    <div
      aria-label="Open mobile menu"
      class="menu_mobile_btn"
      data-testid="mobile-menu-trigger"
      role="button"
    >
      <span />
      <span />
      <span />
    </div>
  </div>
</div>
`;

exports[`Menu Component Mobile View matches snapshot on mobile (menu open) 1`] = `
<div
  class="container"
>
  <div
    class="content content_mobile"
  >
    <img
      alt="Logo"
      class="menu_logo"
      height="35"
      src="/img.jpg"
      width="103"
    />
    <div
      class="modal"
      data-testid="mobile-menu-modal-backdrop"
    />
    <div
      aria-label="Open mobile menu"
      class="menu_mobile_btn"
      data-testid="mobile-menu-trigger"
      role="button"
    >
      <span />
      <span />
      <span />
    </div>
    <div
      class="menu_mobile_box"
      data-testid="mobile-menu-box"
    >
      <span
        aria-label="Close mobile menu"
        class="menu_mobile_close_btn"
        data-testid="mobile-menu-close-button"
        role="button"
      >
        ×
      </span>
      <div
        class="menu_mobile_box_content"
      >
        <div
          class="menu_mobile_nav"
        >
          <ul
            class="menu_mobile_nav_list"
          >
            <li
              class="menu_mobile_nav_list_item"
            >
              <a
                class="menu_mobile_nav_list_item_link"
                href="#"
              >
                Products
              </a>
            </li>
            <li
              class="menu_mobile_nav_list_item"
            >
              <a
                class="menu_mobile_nav_list_item_link"
                href="#"
              >
                Solutions
              </a>
            </li>
            <li
              class="menu_mobile_nav_list_item"
            >
              <a
                class="menu_mobile_nav_list_item_link"
                href="#"
              >
                Pricing
              </a>
            </li>
            <li
              class="menu_mobile_nav_list_item"
            >
              <p
                class="menu_mobile_nav_list_item_link"
                data-testid="mobile-resources-trigger"
                role="button"
              >
                <img
                  alt="seta para esquerda"
                  height="13.34"
                  src="/img.jpg"
                  width="7.67"
                />
                Resources
              </p>
            </li>
            <li
              class="menu_mobile_nav_list_item menu_mobile_nav_list_item_login"
            >
              <a
                class="menu_mobile_nav_list_item_link"
                href="#"
              >
                Log In
              </a>
              <button
                class="menu_desktop_acess_button"
              >
                Sing Up Now
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
